test/config/integration/server.yaml

```diff
@@ -151,7 +151,6 @@ stats_sinks:
   typed_config:
     "@type": type.googleapis.com/envoy.config.metrics.v3.StatsdSink
     tcp_cluster_name: statsd
-watchdog: {}
```

Comment:
This change and the other one like it are unrelated to this change, however when I built with disabling deprecations locally, the hot restart tests failed (we don't compile hot restart in the CI build which tests this), so I just fixed this at the same time. The empty deprecated watchdog config seems a historical legacy and not relevant to hot restart.